src/ucp/rndv/rndv_am.c

```diff
@@ -135,3 +141,89 @@ ucp_proto_t ucp_rndv_am_bcopy_proto = {
+    ucp_request_data_hdr_t hdr;
+    uct_iov_t iov[UCP_MAX_IOV];
+    size_t iov_count;
+
+    ucp_rndv_am_fill_header(&hdr, req);
+
+    ucs_assert(lpriv->super.max_iov > 0);
+    iov_count = ucp_datatype_iter_next_iov(&req->send.state.dt_iter,
+                                           max_payload, lpriv->super.md_index,
+                                           UCP_DT_MASK_CONTIG_IOV, next_iter,
+                                           iov, lpriv->super.max_iov);
+
+    return uct_ep_am_zcopy(ucp_ep_get_lane(req->send.ep, lpriv->super.lane),
+                           UCP_AM_ID_RNDV_DATA, &hdr, hdr_size, iov, iov_count,
+                           0, &req->send.state.uct_comp);
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_rndv_am_zcopy_complete(ucp_request_t *req)
+{
+    ucp_rndv_am_destroy_rkey(req);
+    return ucp_request_invoke_uct_completion_success(req);
+}
+
+static ucs_status_t ucp_proto_rndv_am_zcopy_progress(uct_pending_req_t *uct_req)
+{
+    ucp_request_t *req = ucs_container_of(uct_req, ucp_request_t, send.uct);
+
+    /* coverity[tainted_data_downcast] */
+    return ucp_proto_multi_zcopy_progress(req, req->send.proto_config->priv,
+                                          NULL, UCT_MD_MEM_ACCESS_LOCAL_READ,
+                                          UCP_DT_MASK_CONTIG_IOV,
+                                          ucp_rndv_am_zcopy_send_func,
+                                          ucp_rndv_am_zcopy_complete,
+                                          ucp_proto_request_zcopy_completion);
+}
+
+static ucs_status_t
+ucp_proto_rndv_am_zcopy_init(const ucp_proto_init_params_t *init_params)
+{
+    ucp_context_t *context               = init_params->worker->context;
+    ucp_proto_multi_init_params_t params = {
+        .super.super         = *init_params,
+        .super.cfg_thresh    = context->config.ext.zcopy_thresh,
+        .super.cfg_priority  = 30,
+        .super.min_iov       = 1,
+        .super.min_frag_offs = ucs_offsetof(uct_iface_attr_t, cap.am.min_zcopy),
+        .super.max_frag_offs = ucs_offsetof(uct_iface_attr_t, cap.am.max_zcopy),
+        .super.max_iov_offs  = ucs_offsetof(uct_iface_attr_t, cap.am.max_iov),
+        .super.send_op       = UCT_EP_OP_AM_ZCOPY,
+        .super.memtype_op    = UCT_EP_OP_LAST,
+        .super.flags         = UCP_PROTO_COMMON_INIT_FLAG_SEND_ZCOPY |
+                               UCP_PROTO_COMMON_INIT_FLAG_CAP_SEG_SIZE,
+        .first.tl_cap_flags  = UCT_IFACE_FLAG_AM_ZCOPY,
+        .middle.tl_cap_flags = UCT_IFACE_FLAG_AM_ZCOPY
+    };
+
+    return ucp_proto_rndv_am_init_common(&params);
+}
+
+static void
+ucp_proto_rndv_am_zcopy_abort(ucp_request_t *req, ucs_status_t status)
+{
+    ucp_rndv_am_destroy_rkey(req);
+    ucp_proto_request_zcopy_abort(req, status);
+}
+
+
+ucp_proto_t ucp_rndv_am_zcopy_proto = {
+    .name     = "rndv/am/zcopy",
+    .desc     = UCP_PROTO_ZCOPY_DESC,
+    .flags    = 0,
+    .init     = ucp_proto_rndv_am_zcopy_init,
```

Comment:
I'd like to suggest to update names of the methods.
`ucp_proto_rndv_am_zcopy_init` -> `ucp_rndv_am_zcopy_proto_init`
`ucp_proto_rndv_am_zcopy_progress` -> `ucp_rndv_am_zcopy_proto_progress`
`ucp_proto_rndv_am_zcopy_abort` -> `ucp_rndv_am_zcopy_proto_abort`

Let me explain the suggestion. We had similar comment earlier here https://github.com/openucx/ucx/pull/8019#discussion_r825461677. Then it was addressed in this PR https://github.com/openucx/ucx/pull/8044.

The reason is that we usually use directories and files to create names for the functions, structures, etc.
I know that names used for `ucp_rndv_am_bcopy_proto` have the similar issue. And this can be addressed in the separate PR. The origin of this naming pattern (meaning `ucp_proto`) was to distinguish old and new protocols in common code. However, this file is completely dedicated for new protocol infrastructure. Thus, there is no need to add the prefix.

Sorry for one more single comment.

Everything else looks good to me.